python/pyspark/sql/connect/column.py

```diff
@@ -99,11 +101,59 @@ def to_plan(self, session: Optional["RemoteSparkSession"]) -> "proto.Expression"
         value_type = type(self._value)
         exp = proto.Expression()
         if value_type is int:
-            exp.literal.i32 = cast(int, self._value)
+            exp.literal.i64 = cast(int, self._value)
+        elif value_type is bool:
+            exp.literal.boolean = cast(bool, self._value)
         elif value_type is str:
             exp.literal.string = cast(str, self._value)
         elif value_type is float:
             exp.literal.fp64 = cast(float, self._value)
+        elif value_type is decimal.Decimal:
+            d_v = cast(decimal.Decimal, self._value)
+            v_tuple = d_v.as_tuple()
+            exp.literal.decimal.scale = abs(v_tuple.exponent)
+            exp.literal.decimal.precision = len(v_tuple.digits) - abs(v_tuple.exponent)
+            # Two complement yeah...
+            raise ValueError("Python Decimal not supported.")
+        elif value_type is bytes:
+            exp.literal.binary = self._value
+        elif value_type is datetime.datetime:
+            # Microseconds since epoch.
+            dt = cast(datetime.datetime, self._value)
+            v = dt - datetime.datetime(1970, 1, 1, 0, 0, 0, 0)
+            exp.literal.timestamp = int(v / datetime.timedelta(microseconds=1))
```

Comment:
Can we maybe match the implementation in PySpark's: https://github.com/apache/spark/blob/master/python/pyspark/sql/types.py#L254-L260 ?